src/main/java/org/radarbase/appserver/service/questionnaire/protocol/factory/NotificationHandlerFactory.java

```diff
@@ -24,12 +24,21 @@
 
 import org.radarbase.appserver.service.FcmNotificationService;
 import org.radarbase.appserver.service.questionnaire.protocol.ProtocolHandler;
+import org.radarbase.appserver.service.questionnaire.protocol.DisabledNotificationHandler;
 import org.radarbase.appserver.service.questionnaire.protocol.SimpleNotificationHandler;
+import org.radarbase.appserver.dto.protocol.NotificationProtocol;
 
 public class NotificationHandlerFactory {
 
-    public static ProtocolHandler getNotificationHandler() {
-        return new SimpleNotificationHandler();
+    public static ProtocolHandler getNotificationHandler(NotificationProtocol protocol) {
+        switch (protocol.getMode()) {
+            case STANDARD:
+                return new SimpleNotificationHandler();
+            case DISABLED:
+                return new DisabledNotificationHandler();
```

Comment:
You can add case here for COMBINED and throw a Unsupported/Not implemented error here. So we have a placeholder to add it later